packages/cxl-ui/src/components/cxl-link-button.js

```diff
@@ -0,0 +1,87 @@
+import '@conversionxl/cxl-lumo-styles';
+import { html } from '@polymer/polymer/polymer-element.js';
+import { ButtonElement } from '@vaadin/vaadin-button';
+import { customElement } from 'lit-element';
+
+@customElement('cxl-link-button')
+export class CXLLinkButtonElement extends ButtonElement {
+  static get properties() {
+    return {
+      href: {
+        type: String,
+      },
+    };
+  }
+
+  static get template() {
+    return html`
+      <style>
+        :host {
+          display: inline-block;
+          position: relative;
+          outline: none;
+          white-space: nowrap;
+        }
+
+        :host([hidden]) {
+          display: none !important;
+        }
+
+        /* Ensure the button is always aligned on the baseline */
+        .vaadin-button-container::before {
+          content: '\\2003';
+          display: inline-block;
+          width: 0;
+        }
+
+        .vaadin-button-container {
+          display: inline-flex;
+          align-items: center;
+          justify-content: center;
+          text-align: center;
+          width: 100%;
+          height: 100%;
+          min-height: inherit;
+          text-shadow: inherit;
+          -webkit-user-select: none;
+          -moz-user-select: none;
+          user-select: none;
+        }
+
+        [part='prefix'],
+        [part='suffix'] {
+          flex: none;
+        }
+
+        [part='label'] {
+          white-space: nowrap;
+          overflow: hidden;
+          text-overflow: ellipsis;
+        }
+
+        #button {
+          position: absolute;
+          top: 0;
+          left: 0;
+          right: 0;
+          width: 100%;
+          height: 100%;
+          opacity: 0;
+          cursor: inherit;
+        }
+      </style>
+      <div class="vaadin-button-container">
+        <div part="prefix">
+          <slot name="prefix"></slot>
+        </div>
+        <div part="label">
+          <slot></slot>
+        </div>
+        <div part="suffix">
+          <slot name="suffix"></slot>
+        </div>
+      </div>
+      <a href="[[href]]" id="button" type="button"></a>
```

Comment:
From Slack:

> I'm thinking <a> should be in light DOM maybe for SEO safety?
> As in, we let user nest the full link element, instead of putting href on button (edited)

Reply:
The only way to disable shadow dom per component is to use:

```
_attachDom(dom) {
  this.appendChild(dom);
}
```

This breaks everything from vaadin.

The consensus these days is that major search engines now parse javascript and are able to see shadow dom content as if the shadow dom boundary doesn't exist. This is the case with at least Google and Bing.

Changing to the class implementation wouldn't be difficult, though we would lose some functionality.

Reply:
https://www.searchenginejournal.com/shadow-dom/353644/